src/megaapi_impl.cpp

```diff
@@ -20674,27 +20979,181 @@ void MegaApiImpl::sendPendingRequests()
+
+            e = client->enableSync(tag, syncError, true);
+            //TODO: note to reviewer: in case this fails with a non temporary error, this sets the sync
+            // to FAILED (which oddly entails: isEnabled = true: from the user perspective it has been enabled)
+            // Alternatively, we could keep/enforce a SYNC_DISABLED+NO_SYNC_ERROR (i.e. manually disabled) and
+            //  asume the failure in the req finish is enough for the app and that we do not want to update the sync configuration
+
+            request->setNumDetails(syncError);
+
+            if (!e) //sync added (enabled) fine
+            {
                 fireOnRequestFinish(request, MegaError(API_OK));
+                break;
+            }
+
+            break;
+        }
+        case MegaRequest::TYPE_COPY_SYNC_CONFIG:
+        {
+            const char *localPath = request->getFile();
+            if(!localPath)
+            {
+                e = API_EARGS;
+                break;
+            }
+
+            auto nextSyncTag = client->nextSyncTag(10000);//We artificially start giving syncs number from 10k, to avoid clashing
+            //when mixed syncConfig vs app configs (i.e. new -> old -> new version)
+
+            using CType = CacheableStatus::Type;
+            bool overStorage = client->mCachedStatus[CType::STATUS_STORAGE] ? (client->mCachedStatus[CType::STATUS_STORAGE]->value() >= MegaApi::STORAGE_STATE_RED) : false;
+            bool businessExpired = client->mCachedStatus[CType::STATUS_BUSINESS] ? (client->mCachedStatus[CType::STATUS_BUSINESS]->value() == BIZ_STATUS_EXPIRED) : false;
+            bool blocked = client->mCachedStatus[CType::STATUS_BLOCKED] ? (client->mCachedStatus[CType::STATUS_BLOCKED]->value()) : false;
+
+            auto syncError = NO_SYNC_ERROR;
+            // the order is important here: an user needs to resolve blocked in order to resolve storage
+            if (overStorage)
+            {
+                syncError = STORAGE_OVERQUOTA;
+            }
+            else if (businessExpired)
+            {
+                syncError = BUSINESS_EXPIRED;
+            }
+            else if (blocked)
+            {
+                syncError = ACCOUNT_BLOCKED;
+            }
+
+            bool enabled = request->getFlag();
+            bool temporaryDisabled = request->getNumDetails();
+
+            if (!enabled && temporaryDisabled)
+            {
+                if (!syncError) syncError = UNKNOWN_TEMPORARY_ERROR;
+                enabled = true; //we consider enabled when it is temporary disabled
+            }
+
+            SyncConfig syncConfig{nextSyncTag, localPath, request->getNodeHandle(),
+                                  static_cast<fsfp_t>(request->getNumber()),
+                                  regExpToVector(request->getRegExp()), enabled};
+
+            if (temporaryDisabled)
+            {
+                if (!syncError)
+                {
+
```

Comment:
??

Maybe this block was moved to the one inside the block for `if (!enabled && temporaryDisabled)`? If so, maybe the error can be set in there for the `syncConfig` (and create it earlier :) )

Reply:
exactly, it was moved there. SyncConfig ctor takes enabled. can't move it before. But I'll remove the empty block